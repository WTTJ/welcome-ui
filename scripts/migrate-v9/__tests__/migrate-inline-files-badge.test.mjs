import { readFileSync } from 'fs'
import { resolve } from 'path'

import { describe, expect, it } from 'vitest'

import { setupMigrationTest } from '../helpers/test-setup.mjs'
import { migrateAll } from '../index.mjs'

describe('Inline Migration - Badge Components', () => {
  const tempDir = setupMigrationTest('inline-badge-test', '__fixtures__')

  it('migrates Badge components with styling props to Badge with classNames', async () => {
    // Run migration
    await migrateAll(tempDir, { copyDir: false, interactive: false })

    // Read results
    const migratedComponent = readFileSync(resolve(tempDir, 'BadgeComponent.tsx'), 'utf8')

    // Verify with snapshots
    expect(migratedComponent).toMatchSnapshot('badge-component.tsx')
  })
})
