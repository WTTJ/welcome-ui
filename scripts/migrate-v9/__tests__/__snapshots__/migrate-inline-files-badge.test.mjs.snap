// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Inline Migration - Badge Components > migrates Badge components with styling props to Badge with classNames > badge-component.tsx 1`] = `
"import { Badge } from 'welcome-ui/Badge'

export const BadgeComponent = () => {
  return (
    <div>
      <Badge className="ml-sm" size="sm" variant="primary">
        14
      </Badge>
      <Badge className="mr-md" size="lg" variant="secondary">
        New
      </Badge>
    </div>
  )
}
"
`;
